pymc3/sampling.py

```diff
@@ -427,8 +427,10 @@ def sample(
         check_start_vals(model.test_point, model)
     else:
         if isinstance(start, dict):
+            start = {k: v for k, v in start.items()}
             update_start_vals(start, model.test_point, model)
         else:
+            start = start[:]
```

Comment:
This will still change the dictionary inplace.

```python
a = [dict(a=1, b=2), dict(a=1, b=2)]
b = a[:]
for b_ in b:
    b_['c'] = 3
    
a
[{'a': 1, 'b': 2, 'c': 3}, {'a': 1, 'b': 2, 'c': 3}]
```

Reply:
You can try `b = [a_dict.copy() for a_dict in a]`

Reply:
yeah, I think @chandan5362 original suggestion to use deepcopy was good

Reply:
Since the function in question only adds new keys, but does not change keys already in place, a shallow copy (and nested shallow copy) should be fine. But I have no strong objections to deepcopy

Reply:
yea, we could use `start.copy()` but to stay on the safer side, we should probaly use `deepcopy` though it does not make any sense here. Also we won't have to use `start.copy` inside `list comprehension` if we use `deepcopy`.

Reply:
You're right, `deepcopy(None) == None`, so it can go even before any `is not None` or `isinstance` checks.

Reply:
This line can also be removed now.